docs/en/advanced_tutorials/config.md

```diff
@@ -1,3 +1,579 @@
+```
+
+Json：
+
+```json
+{
+  "test_int": 1,
+  "test_list": [1, 2, 3],
+  "test_dict": {"key1": "value1", "key2": 0.1}
+}
+```
+
+YAML：
+
+```yaml
+test_int: 1
+test_list: [1, 2, 3]
+test_dict:
+  key1: "value1"
+  key2: 0.1
+```
+
+For the above three formats, assuming the file names are `config.py`, `config.json`, and `config.yml`. Loading these files with `Config.fromfile('config.xxx')` will return the same result, which contain `test_int`, `test_list` and `test_dict` 3 variables.
+
+Let's take `config.py` as an example:
+
+```python
+from mmengine.config import Config
+
+cfg = Config.fromfile('learn_read_config.py')
+print(cfg)
+```
+
+```
+Config (path: learn_read_config.py): {'test_int': 1, 'test_list': [1, 2, 3], 'test_dict': {'key1': 'value1', 'key2': 0.1}}
+```
+
+## How to use `Config`
+
+After loading the configuration file, we can access the data stored in `Config` instance just like getting/writing value from `dict`, or getting/writing attribute from instance.
+
+```python
+print(cfg.test_int)
+print(cfg.test_list)
+print(cfg.test_dict)
+cfg.test_int = 2
+
+print(cfg['test_int'])
+print(cfg['test_list'])
+print(cfg['test_dict'])
+cfg['test_list'][1] = 3
+print(cfg['test_list'])
+```
+
+```
+1
+[1, 2, 3]
+{'key1': 'value1', 'key2': 0.1}
+2
+[1, 2, 3]
+{'key1': 'value1', 'key2': 0.1}
+[1, 3, 3]
+```
+
+```{note}
+The `dict` object parsed by `Config` will be converted to `ConfigDict`, and then we can access the value of the `dict` the same as accessing the attribute of a instance.
+```
+
+We can use the `Config` combination with the [Registry](./registry.md) to build registered instance easily.
+
+Here is an example of defining optimizers in a configuration file.
+
+`config_sgd.py`：
+
+```python
+optimizer = dict(type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001)
+```
+
+Suppose we have defined a registry `OPTIMIZERS`, which includes various optimizers, and then We can build the optimizer as below
+
+```python
+from mmengine import Config, optim
+from mmengine.registry import OPTIMIZERS
+
+import torch.nn as nn
+
+cfg = Config.fromfile('config_sgd.py')
+
+model = nn.Conv2d(1, 1, 1)
+cfg.optimizer.params = model.parameters()
+optimizer = OPTIMIZERS.build(cfg.optimizer)
+print(optimizer)
+```
+
+```
+SGD (
+Parameter Group 0
+    dampening: 0
+    foreach: None
+    lr: 0.1
+    maximize: False
+    momentum: 0.9
+    nesterov: False
+    weight_decay: 0.0001
+)
+```
+
+## Inheritance between configuration files
+
+Sometimes, the difference between two different configuration files is so small that only one field may be changed. However, we need to copy and paste everything, and only modify one line. It is not easy to locate the specific difference when we go back to this after a long time.
+
+In another case, multiple configuration files may have the same batch of fields, and we have to copy and paste them in different configuration files. It will also be hard to maintain these fields in a long time.
```

Comment:
```suggestion
In another case, multiple configuration files may have the same batch of fields, and we have to copy and paste them in different configuration files. It will also be hard to maintain these fields in a long time.

We address these issues with inheritance mechanism, detailed as below.
```